Form validation: Improve demo pages

EricDunsworth opened the pull request:
Overview of changes...
- Both:
  - Fix prettify implementations.
  - Use strong elements (instead of b) for bolded note box text.
  - Remove problematic class="required" attributes from certain checkbox labels.
  - Make code indentation more consistent (by always placing labels with child inputs on the same line).
  - Remove empty lines from start/end of code samples.
  - Update second reset button to match the first one.
- English:
  - Add " (including area code)" to the phone number label's code sample. It was already present in the phone number's demo.
- French:
  - Place metadata description on a single line.
  - Port some English-only changes from #8744 (i.e. plural examples heading, second section and its H2 heading).
  - Rename "empilés" to "verticaux".
  - Add "1" to the end IDs/names in certain demos to make them consistent with their English counterparts.
  - Add " (avec l'indicatif régional)" to the phone number label. It was already present in the phone number's code sample.
  - Reduce indentation of number field's code sample.

EricDunsworth commented on src/plugins/formvalid/formvalid-fr.hbs:
```diff
@@ -448,6 +428,13 @@
+<hr>
+<section>
+<!-- checkbox in list item -->
+	<h2>Exemples de formulaires empilés et horizontaux utilisant des éléments de liste comme contenants</h2>
```
I improvised this heading's content.

duboisp replied:
Instead of "empilés" you can say "formulaires verticaux et horizontaux"

EricDunsworth replied:
Done. Also applied it to other instances of "empilés".

EricDunsworth commented on src/plugins/formvalid/formvalid-en.hbs:
```diff
@@ -759,12 +676,10 @@
 			</details>
 
-			<input type="submit" value="Submit" class="btn btn-primary" /> <input type="reset" value="Reset" class="btn btn-default" />
+			<input type="submit" value="Submit" class="btn btn-primary" /> <input type="reset" value="Reset page to defaults" class="btn btn-link btn-sm show p-0 pt-4" />
```
@duboisp @delisma Was this reset button intentionally meant to resemble the older reset button design?

GormFrank reviewed (approved):
The reset button will need to be fixed in a different PR.

duboisp commented:
Approved at the technical review

EricDunsworth committed:
Form validation: Improve demo pages

Overview of changes...
- Both:
  - Fix prettify implementations.
  - Use strong elements (instead of b) for bolded note box text.
  - Remove problematic class="required" attributes from certain checkbox labels.
  - Make code indentation more consistent (by always placing labels with child inputs on the same line).
  - Remove empty lines from start/end of code samples.
  - Update second reset button to match the first one.
- English:
  - Add " (including area code)" to the phone number label's code sample. It was already present in the phone number's demo.
- French:
  - Place metadata description on a single line.
  - Port some English-only changes from #8744 (i.e. plural examples heading, second section and its H2 heading).
  - Rename "empilés" to "verticaux".
  - Add "1" to the end IDs/names in certain demos to make them consistent with their English counterparts.
  - Add " (avec l'indicatif régional)" to the phone number label. It was already present in the phone number's code sample.
  - Reduce indentation of number field's code sample.

EricDunsworth commented:
@GormFrank @duboisp Should be good to merge now :).
